### PR TITLE
feat(sync): 将 Grok Build 用量记入项目用量

### PR DESCRIPTION
**Pull request by Roomen**
## Summary

仪表盘「项目用量」统计不到 Grok Build。Grok 总用量已经进 `queue.jsonl`，但解析器从未写入 `project.queue.jsonl`，所以无法挂到仓库项目上。

这次把 `parseGrokBuildIncremental` 接到现有的 Claude/Codex/Droid/Omp 项目用量链路，并用独立的项目 cursor，已同步过的安装可以回填历史，而不会把总用量再记一遍。

## Scope

- [x] CLI (`src/`)
- [ ] Dashboard (`dashboard/`)
- [ ] macOS app (`TokenTrackerBar/`)
- [ ] Windows app (`TokenTrackerWin/`)
- [ ] Docs / CI / config

## Checklist

- [x] `node --test test/grok-parser.test.js`（11 pass）以及原有 Grok 回归测试（15 pass）
- [x] 无新增用户文案
- [x] commit 仍为 conventional English（`feat:` / `fix:` / …）
- [x] PR 说明写了原因，不只是改了什么

## 原因

项目用量只读 `project.queue.jsonl`。Grok session 其实已有可还原的工作目录（`summary.info.cwd`、URL 编码的 session 目录、SessionEnd hook 的 `cwd`），解析器没用。会话页能看到文件夹名，项目用量面板看不到。

## 行为

- 工作目录顺序：`summary.info.cwd` → hook `cwd` → `decodeURIComponent(encodedCwd)` → session 目录名
- 仅当 `projectKey && projectRef`（`public_verified`）才入队，隐私门槛与 Claude/Codex 相同
- 没有 git / 未验证的仓库不推进项目 cursor，之后还可以重试
- `legacyBaselineOnly` 只压总用量，不压项目回填
- `GROK_CURSOR_VERSION` 保持 4（不触发总用量重建）
- auto / 增量 sync 如果只写出项目用量，也会提交 cursor，避免下一轮重复入队
- 已删除 session 的 `projectUpdateOffsets` 会裁掉；仍在磁盘上的未验证 session 保留

## 测试

- 升级回填不会重写总用量
- 重复 sync 幂等
- 增量只追加 delta
- 没有 `summary.info.cwd` 时仍可用 encodedCwd
- 无 git：不写项目行；补上 `.git` 后能回填
- 无 `turn_completed` 时用水印快照，且与总用量 cursor 独立
- 删除 session 后裁剪项目 offset
- sync.js 在仅有项目桶时不会 skip cursor 提交

<details>
<summary><strong>Codex review context</strong></summary>

- **Delta since last Codex review:** `c5c73865` 修了项目-only auto sync 丢 cursor，以及已删 session 的 offset 膨胀
- **Intended behavior / invariants:** 不改变 Grok 总用量计数；回填不得再追加 `queue.jsonl`；只存储 `public_verified` 的 git remote
- **Edge cases covered:** 升级回填、幂等、增量、encodedCwd-only、无 git 重试、watermark、offset 裁剪、project-only cursor 提交
- **Tests run (command + result):** `node --test test/grok-parser.test.js` 11 pass；`node --test --test-name-pattern "parseGrokBuildIncremental|Grok" test/rollout-parser.test.js` 14 pass
- **Known gaps / out of scope:** 仪表盘/API 未改（已按 `row.source` 聚合）；未验证 session 每次 sync 重读是故意的

### Most likely regression surface

- Grok 全局 cursor / v4 turn-usage 迁移
- 项目 cursor 若绑到 `updateOffsets` 会双计

### Verification method (choose at least one)

- 上述解析器测试

### Uncovered scope

- 未跑完整 `npm test`
- 未做仪表盘点击验证（无 UI 改动）

</details>

### CODE REVIEW

File: src/lib/rollout.js [hunk 1]
```diff
@@ -15721,6 +15757,25 @@ async function parseGrokBuildIncremental({
...
+  const projectMetaCache = projectEnabled ? new Map() : null;
+  const publicRepoCache = projectEnabled ? new Map() : null;
+  // Independent of global updateOffsets so an upgrade can backfill historical
+  // turn_completed events into project.queue.jsonl without re-appending totals.
+  const projectUpdateOffsets =
+    projectEnabled &&
+    grokState.projectUpdateOffsets &&
+    typeof grokState.projectUpdateOffsets === "object"
+      ? { ...grokState.projectUpdateOffsets }
+      : {};
```
[the remark this answers is not in the document]
Comment (Roomen) [reply src/lib/rollout.js:15787]: Addressed in c5c73865.

`projectUpdateOffsets` is now pruned by `existsSync` on the `updates.jsonl` path (`pruneMissingGrokProjectUpdateOffsets`). Deleted session files drop out; still-present unverified/blocked sessions keep their keys so a later public-verified retry is possible. Filtering to “successfully attributed this scan” would have dropped those retryable entries.

**Comment by Roomen** [comment]
Addressed the two CodeRabbit majors in c5c73865:

1. `projectBucketsQueued` is merged into `grokResult` and included in `skipNoOpCursorCommit`, so a project-only auto/incremental backfill cannot write `project.queue.jsonl` while discarding the project cursor.
2. `projectUpdateOffsets` drops keys whose `updates.jsonl` no longer exists, without pruning still-on-disk unverified sessions.

Nits (tmpdir cleanup, docstring coverage) left as-is.
